### PR TITLE
Added stateful behaviour to Redlining Components

### PR DESCRIPTION
**Pull request by weskamm**
This PR adds some stateful behaviour to Redlining with the help of `getState` and `setState` methods

### CODE REVIEW

File: src/util/Color.js [hunk 1]
```diff
@@ -32,6 +36,23 @@ Ext.define('BasiGX.util.Color', {
...
+         *     // hex is now: "#ff0000"
+         */
+        rgbaToHex: function(rgba){
+            var regex = new RegExp("^rgba?[\\s+]?\\([\\s+]?(\\d+)[\\s+]?," +
+                "[\\s+]?(\\d+)[\\s+]?,[\\s+]?(\\d+)[\\s+]?", "i");
+            rgba = rgba.match(regex);
+            return (rgba && rgba.length === 4) ? "#" +
+                ("0" + parseInt(rgba[1],10).toString(16)).slice(-2) +
+                ("0" + parseInt(rgba[2],10).toString(16)).slice(-2) +
+                ("0" + parseInt(rgba[3],10).toString(16)).slice(-2) : '';
```
Comment (marcjansen) [inline src/util/Color.js]: So we will loose the `a`, right? We should document that.

File: src/view/container/RedlineStyler.js [hunk 2]
```diff
@@ -655,5 +721,67 @@ Ext.define("BasiGX.view.container.RedlineStyler", {
...
+        return state;
+    },
+
+    /**
+     * Method set the state of the redlining styles
+     */
+    setState: function(state) {
+        var me = this;
+        var redliningContainer = Ext.ComponentQuery.query(
+            'basigx-container-redlining')[0];
```
Comment (marcjansen) [inline src/view/container/RedlineStyler.js]: Do we need a `ComponentQuery` here?

**Comment by marcjansen** [comment]
Looks good to me, nice work.

Please rebase this and then merge with or without taking my comments.

File: src/util/Color.js (hunk 1 above)
Comment (weskamm) [inline src/util/Color.js]: Comment added

File: src/view/container/RedlineStyler.js (hunk 2 above)
Comment (weskamm) [inline src/view/container/RedlineStyler.js]: Yes

**Comment by weskamm** [comment]
Documentation added and rebased, will merge when green

**Pull request merged**